libs/git/clone.go

```diff
@@ -0,0 +1,72 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// source: https://stackoverflow.com/questions/59081778/rules-for-special-characters-in-github-repository-name
+var githubRepoRegex = regexp.MustCompile(`^[\w-\.]+$`)
+
+const githubUrl = "https://github.com"
+const databricksOrg = "databricks"
+
+type cloneOptions struct {
+	// Branch or tag to clone
+	Reference string
+
+	// URL for the repository
+	RepositoryUrl string
+
+	// Local path to clone repository at
+	TargetPath string
+}
+
+func (opts cloneOptions) args() []string {
+	args := []string{"clone", opts.RepositoryUrl, opts.TargetPath, "--depth=1", "--no-tags"}
+	if opts.Reference != "" {
+		args = append(args, "--branch", opts.Reference)
+	}
+	return args
+}
+
+func Clone(ctx context.Context, url, reference, targetPath string) error {
```

Comment:
The argument suggests `reference` but the function above and test suggest it can only be a branch.

Which one applies?

Reply:
The `--branch` flag can take in both tags and branches.

source: https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---branchltnamegt